autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/SelectRatesClassification.py

```diff
@@ -0,0 +1,123 @@
+            raise ValueError("score_func must be in ('chi2, 'f_classif', 'mutual_info_classif'), "
+                             "but is: %s" % score_func)
+
+        super().__init__(random_state=random_state)
+        self.add_fit_requirements([
+            FitRequirement('issparse', (bool,), user_defined=True, dataset_property=True),
+            FitRequirement('issigned', (bool,), user_defined=True, dataset_property=True)])
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseEstimator:
+
+        self.check_requirements(X, y)
+
+        self.preprocessor['numerical'] = GenericUnivariateSelect(
+            mode=self.mode, score_func=self.score_func, param=self.alpha)
+
+        return self
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None,
+        alpha: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="alpha",
+                                                                     value_range=(0.01, 0.5),
+                                                                     default_value=0.1,
+                                                                     ),
+        mode: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="mode",
+                                                                    value_range=('fpr', 'fdr', 'fwe', "percentile"),
+                                                                    default_value='fpr',
+                                                                    ),
+        score_func: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="score_func",
+                                                                          value_range=("chi2",
+                                                                                       "f_classif",
+                                                                                       "mutual_info_classif"),
+                                                                          default_value="chi2",
+                                                                          ),
+    ) -> ConfigurationSpace:
+        value_range = list(score_func.value_range)
+        if dataset_properties is not None:
+            if (
+                dataset_properties.get("issigned") is True
+            ):
+                value_range = [value for value in value_range if value not in ("chi2", "mutual_info_classif")]
+            if dataset_properties.get("issparse") is True:
+                value_range = [value for value in value_range if value != "f_classif"]
+
+        if value_range != list(score_func.value_range):
+            warnings.warn(f"Given choices for `score_func` are not compatible with the dataset. "
+                          f"Updating choices to {value_range}")
+
+        if len(value_range) == 0:
+            raise TypeError("`SelectRatesClassification` is not compatible with the"
```

Comment:
TypeError is used when `type(something)` is not appropriate.
```suggestion
            raise ValueError("`SelectRatesClassification` is not compatible with the"
```